scoring/src/main/scala/ai/h2o/sparkling/ml/models/HasMojoDataV2.scala

```diff
@@ -0,0 +1,46 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package ai.h2o.sparkling.ml.models
+
+import java.io.ByteArrayInputStream
+
+import ai.h2o.mojos.runtime.MojoPipeline
+import ai.h2o.mojos.runtime.readers.MojoPipelineReaderBackendFactory
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.sql.SparkSession
+
+private[models] trait HasMojoDataV2 extends HasMojoData {
+
+  private def loadMojoPipeline(mojoData: Array[Byte]): Broadcast[MojoPipeline] = {
+    val reader = MojoPipelineReaderBackendFactory.createReaderBackend(new ByteArrayInputStream(mojoData))
+    val pipeline = MojoPipeline.loadFrom(reader)
+    SparkSession.builder().getOrCreate().sparkContext.broadcast(pipeline)
```

Comment:
@jakubhava Please can you share details why broadcasting of `MojoPipeline` would be faster tham broadcasting of a byte array? Won't the mojo pipeline be serialized and deserialized again?

Reply:
The issue was that we had to create MOJOPipeline instance all over again from bytes which takes time. I want to benchmark it in a bit to see whether this has some impact or not. But yup, it will be serialized and deserialized -> but all the time:

> Broadcast variables allow the programmer to keep a read-only variable cached on each machine rather than shipping a copy of it with tasks.

From now https://jaceklaskowski.gitbooks.io/mastering-apache-spark/spark-broadcast.html, that seems like exactly something we are looking for

Reply:
I know that you were trying multiple implementations in the past, but couldn't creation of `MOJOPipeline` instance for each thread avoided via `lazy val` on a object?

Reply:
I believe the advantage of this approach is that we create the instance just once and transfer it to each executor just once. We do not need to even create new instance per thread. With bi g mojos, creating mojo pipeline takes long time.

I will do some benchmarks and post them here to see what impact (or not) this change has